libsplinter/src/service/timer_alarm_factory.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains `TimerAlarmFactory` trait.
+
+use super::TimerAlarm;
+
+/// Used to create new `TimerAlarm` instances.
+pub trait TimerAlarmFactory: Send {
+    /// Returns a new `TimerAlarm`
+    fn new_alarm(&self) -> Box<dyn TimerAlarm>;
+
+    fn clone_box(&self) -> Box<dyn TimerAlarmFactory>;
```

Comment:
Is this clone_box necessary? I couldn't find where its used in subsequent commits.

Reply:
It is only used directly in the impl of Clone below. That is required because the ScabbardMessageHandlerFactory, which will get a boxed TimerAlarmFactory, must be cloneable.